### PR TITLE
feat(Dialog): add prop `contentOverflow` to control overflow behaviour

### PR DESCRIPTION
**Pull request by DenisVershkov**

### CODE REVIEW

**Comment by DenisVershkov** [comment]
[DATAUI-1254]

**Comment by korvin89** [comment]
Could u provide some description?

**Comment by DenisVershkov** [comment]
1) Added prop 'scroll' to the Modal and Dialog. The prop handles scroll appearance:  
- if 'inner' is selected, then scroll will be applied to the DialogBody
- if 'outer' is selected, then scroll will appear outside (the same behaviour as before)
- 'outer' is the default value

2) Added prop 'withDivider' to the DialogBody component. When this prop has 'true' value, then the DialogBody itself will have top and bottom border. This is made for the 'inner' scroll case and should be used something like: <DialogBody withDivider={scroll === 'inner'}>

File: src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ interface DialogOwnProps {
     disableFocusTrap?: boolean;
     disableAutoFocus?: boolean;
     restoreFocusRef?: React.RefObject<HTMLElement>;
+    scroll?: 'inner' | 'outer';
```
Comment (amje) [inline src/components/Dialog/Dialog.tsx:41]: I suggest this naming instead: `contentOverflow?: 'visible' | 'auto'`, symmetrical to the CSS overflow property

File: src/components/Modal/Modal.scss [hunk 2]
```diff
@@ -102,4 +121,5 @@ $block: '.#{variables.$ns}modal';
 :root {
     --yc-modal-border-radius: 5px;
     --yc-modal-margin: 20px;
+    --yc-modal-with-inner-scroll-margin: 24px;
```
Comment (amje) [inline src/components/Modal/Modal.scss:124]: From v6 we introduce public CSS API, this looks like a private. Please use SCSS var or name it like `--_--inner-scroll-margin` on component's root, not `:root`

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Dialog/Dialog.scss [hunk 3]
```diff
@@ -5,6 +5,13 @@ $block: '.#{variables.$ns}dialog';
 #{$block} {
     position: relative;
 
+    &_with-scroll {
+        display: flex;
+        flex-direction: column;
```
Comment (amje) [inline src/components/Dialog/Dialog.scss:10]: Let's make container always flex, not with this mod only

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -129,8 +131,8 @@ export function Modal({
                 }}
             >
                 <div ref={containerRef} style={style} className={b({open}, className)} data-qa={qa}>
-                    <div className={b('table')}>
-                        <div className={b('cell')}>
+                    <div className={b('content-aligner-wrapper')}>
+                        <div className={b('content-container')}>
```
Comment (amje) [inline src/components/Modal/Modal.tsx:135]: Let's call them `content-aligner` and `content-wrapper`

File: src/components/Dialog/DialogBody/DialogBody.tsx [hunk 5]
```diff
@@ -9,10 +9,11 @@ const b = block('dialog-body');
 export interface DialogBodyProps {
     children: React.ReactNode;
     className?: string;
+    withDividers?: boolean;
 }
 
 export function DialogBody(props: DialogBodyProps) {
-    const {className} = props;
+    const {className, withDividers = false} = props;
```
Comment (amje) [inline src/components/Dialog/DialogBody/DialogBody.tsx:16]: We use `has*` naming throughout the codebase (i.e. `hasClear`, `hasClose`, `hasArrow`)

File: src/components/Dialog/Dialog.scss [hunk 6]
```diff
@@ -4,6 +4,13 @@ $block: '.#{variables.$ns}dialog';
 
 #{$block} {
     position: relative;
+    display: flex;
+    flex-direction: column;
+
+    &_with-scroll {
+        overflow-y: auto;
+        max-height: calc(100vh - var(--yc-modal-margin) * 2);
```
Comment (amje) [inline src/components/Dialog/Dialog.scss:12]: Let's use SCSS var instead, this var will be removed in v6

**Review by amje — CHANGES_REQUESTED** [review]

**Comment by DenisVershkov** [comment]
in order not to deal with merge conflicts, i created a new one #1251

**Comment by DenisVershkov** [comment]
in order not deal with merge conflicts, i created a new one #1251

**Pull request closed without merging**